project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/sample/SampleRepositoryImpl.java

```diff
@@ -87,4 +122,54 @@ public Result<Collection<Sample>, SampleInformationService.ResponseCode> findSam
     return Result.fromValue(samples);
   }
 
+  @Override
+  public Result<Collection<Sample>, SampleInformationService.ResponseCode> findSamplesByBatchId(
+      BatchId batchId) {
+    Objects.requireNonNull(batchId);
+    Collection<Sample> samples;
+    try {
+      samples = qbicSampleRepository.findAllByAssignedBatch(batchId);
+    } catch (Exception e) {
+      log.error(
+          "Retrieving Samples for batch with id " + batchId.value() + " failed: " + e, e);
+      return Result.fromError(SampleInformationService.ResponseCode.QUERY_FAILED);
+    }
+    return Result.fromValue(samples);
+  }
+
+  @Transactional
+  @Override
+  public Result<Collection<Sample>, ResponseCode> updateAll(Project project,
+      Collection<Sample> updatedSamples) {
+    Collection<SampleId> sampleIds = new ArrayList<>();
+    updatedSamples.forEach(sample -> sampleIds.add(sample.sampleId()));
+    String commaSeperatedSampleIds = sampleIds.stream().map(Object::toString).collect(
+        Collectors.joining(", "));
+    try {
+      this.qbicSampleRepository.saveAll(updatedSamples);
```

Comment:
here, a result object makes sense. When the save action fails, you will likely try to fix the issue and try again or request further information,...